consumer.go

```diff
@@ -547,9 +548,15 @@ func (child *partitionConsumer) parseRecords(batch *RecordBatch) ([]*ConsumerMes
 		}
 	}
 
-	if incomplete || len(messages) == 0 {
+	if incomplete {
 		return nil, ErrIncompleteResponse
 	}
+
+	child.offset = batch.FirstOffset + int64(batch.LastOffsetDelta) + 1
+	if child.offset == originalOffset {
```

Comment:
`<=`?

Reply:
Not sure if this would happen but it is definitely better to check this case. I've the operator to ```<=``` now.